prober/icmp.go

```diff
@@ -136,3 +132,96 @@ func ProbeICMP(ctx context.Context, target string, module config.Module, registr
 		}
 	}
 }
+
+func getICMPSocket(
+	cfg config.ICMPProbe,
+	ip *net.IPAddr,
+	registry *prometheus.Registry,
+	logger log.Logger) (socket net.PacketConn, requestType icmp.Type, replyType icmp.Type, err error) {
+
+	level.Info(logger).Log("msg", "Creating socket")
+
+	if ip.IP.To4() == nil {
+		requestType = ipv6.ICMPTypeEchoRequest
+		replyType = ipv6.ICMPTypeEchoReply
+		socket, err = icmp.ListenPacket("ip6:ipv6-icmp", "::")
+	} else {
+		requestType = ipv4.ICMPTypeEcho
+		replyType = ipv4.ICMPTypeEchoReply
+
+		if !cfg.DontFragment {
+			socket, err = icmp.ListenPacket("ip4:icmp", "0.0.0.0")
+			return
+		}
+
+		s, err := net.ListenPacket("ip4:icmp", "0.0.0.0")
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		rc, err := ipv4.NewRawConn(s)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		socket = &dfConn{c: rc}
+	}
+
+	return
+}
+
+type dfConn struct {
+	c *ipv4.RawConn
+}
+
+func (c *dfConn) ReadFrom(b []byte) (int, net.Addr, error) {
+	h, p, _, err := c.c.ReadFrom(b)
```

Comment:
This won't work on Windows, which is a supported platform.

Reply:
How do you propose I move about this? Different files for nix and windows? Or error when reading config?

Reply:
First see if this can be done with Windows, we should be aiming for feature parity